test/emu_base/math/test_krylov_exp.py

```diff
@@ -185,23 +188,80 @@ def test_converged_non_hermitian_non_normalized():
     )
 
 
+def test_krylov_with_matrix():
+
```

Comment:
you use matrix sigma_X and others several times.
You can define it one in the beginning only and reuse it